File: SOAP/clisoap1.php
<?php
// Definición del archivo descriptor del servicio web remoto
$wsdl = "https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL";

// Creación del cliente SOAP nativo usando la extensión previamente activada
$cliente = new SoapClient($wsdl);

// Ejecución de la función remota enviando el parámetro 'n' capturado de la URL
$respuesta = $cliente->NumberToWords(array("ubiNum" => $_GET['n']));

// Impresión del resultado directo (el número traducido al idioma inglés por defecto)
echo $respuesta->NumberToWordsResult;
?>

File: SOAP/clisoap1.rb
require 'webrick'
require 'soap/wsdlDriver'

server = WEBrick::HTTPServer.new(
  Port: 8001
)

server.mount_proc '/' do |req, res|

  numero = req.query['n'] || "0"

  wsdl = "https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL"

  cliente = SOAP::WSDLDriverFactory.new(wsdl).create_rpc_driver

  respuesta = cliente.NumberToWords(
    {"ubiNum"=>numero.to_i}
  )

  res.body = respuesta.number_to_words_result

end

trap("INT"){server.shutdown}

puts "Servidor SOAP activo puerto 8001"

server.start